### PR TITLE
Feature/205 auth

### LINKED ISSUE #205: Send user+password to backend and check if users is logged in
Needs https://github.com/yaptide/yaptide/issues/92

### PR DESCRIPTION
**Pull request by ostatni5**
close #205

### CODE REVIEW

**Review by Derstilon — CHANGES_REQUESTED** [review]
I have few questions you can find in the comments

File: src/services/AuthService.tsx [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+	refresh: () => void;
+	status: () => void;
+	authKy: KyInstance;
+}
+
+const [useAuth, AuthContextProvider] = createGenericContext<IAuth>();
+
+const Auth = (props: AuthProps) => {
+	const [user, setUser] = useState<AuthUser | null>(load(StorageKey.USER));
+	const [refreshInterval, setRefreshInterval] = useState(1000 * 60 * 3);
```
[the remark this answers is not in the document]
Comment (ostatni5) [reply src/services/AuthService.tsx:45]: Well, I think that value is more readable as 3 mins than 180000.

[the remark this answers is not in the document]
Comment (ostatni5) [reply src/services/AuthService.tsx:45]: It is overrated by backend response anyway, so it is not so important.

[the remark this answers is not in the document]
Comment (Derstilon) [reply src/services/AuthService.tsx:45]: Ok numbers can be left the way they are now. Just add simple comment explaining the numbers. Something like `// default refresh interval: 3 min`

File: src/services/AuthService.tsx [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+	name: string;
+}
+
+const load = (key: string) => {
+	const item = localStorage.getItem(key) ?? 'null';
+	const obj = JSON.parse(item.length === 0 ? 'null' : item);
+	return obj;
+};
+
+const save = (key: string, value: any) => {
```
Comment (Derstilon) [inline src/services/AuthService.tsx:22]:
```suggestion
const save = (key: string, value: unknown) => {
```

File: src/services/AuthService.tsx [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+};
+
+enum StorageKey {
+	USER = 'editor.user'
+}
+
+export interface IAuth {
+	user: AuthUser | null;
+	isAuthorized: boolean;
+	login: (username: string, password: string) => void;
```
[the remark this answers is not in the document]
Comment (ostatni5) [reply src/services/AuthService.tsx:34]: Yes, we are using username and password to authorize.

File: src/WrapperApp/components/LoginPanel.tsx [hunk 4]
```diff
@@ -1,19 +1,20 @@
...
+import React, { useState } from 'react';
+import { useAuth } from '../../services/AuthService';
 
-interface LoginPanelProps {
-	handleLogin: (data: { email: string; password: string }) => void;
-}
+export default function LoginPanel() {
+	const { login } = useAuth();
+
+	const [email, setEmail] = useState('');
```
Comment (Derstilon) [inline src/WrapperApp/components/LoginPanel.tsx:8]:
```suggestion
	const [username, setUsername] = useState('');
```

File: src/WrapperApp/components/LoginPanel.tsx [hunk 5]
```diff
@@ -47,13 +48,7 @@ export default function LoginPanel(props: LoginPanelProps) {
...
 					onChange={handlePasswordChange}
 				/>
-				<Button
-					variant='outlined'
-					onClick={() => {
-						setPassword('');
-						setEmail('');
-						props.handleLogin({ email, password });
-					}}>
+				<Button variant='outlined' onClick={() => login(email, password)}>
```
Comment (Derstilon) [inline src/WrapperApp/components/LoginPanel.tsx:51]:
```suggestion
				<Button variant='outlined' onClick={() => login(username, password)}>
```

File: src/WrapperApp/components/LoginPanel.tsx [hunk 6]
```diff
@@ -1,19 +1,20 @@
...
+export default function LoginPanel() {
+	const { login } = useAuth();
+
+	const [email, setEmail] = useState('');
+	const [password, setPassword] = useState('');
 
-export default function LoginPanel(props: LoginPanelProps) {
-	const [email, setEmail] = React.useState('');
 	const handleEmailChange = (event: React.ChangeEvent<HTMLInputElement>) => {
 		setEmail(event.target.value);
```
Comment (Derstilon) [inline src/WrapperApp/components/LoginPanel.tsx:12]:
```suggestion
	const handleUsernameChange = (event: React.ChangeEvent<HTMLInputElement>) => {
		setUsername(event.target.value);
```

**Review by Derstilon — APPROVED** [review]
Good job!

**Pull request merged**